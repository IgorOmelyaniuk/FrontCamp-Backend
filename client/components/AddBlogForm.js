import React from 'react'

class AddBlog extends React.Component {

    addBlogHandler = (event) => {
        event.preventDefault()
        const data = {
            title: this.titleInput.value,
            text: this.textInput.value,
            author: this.authorInput.value
        }
        this.props.addBlog(data);
        this.refreshInput(this.titleInput, this.textInput, this.authorInput);
    }

    refreshInput = (...inputs) => {
        inputs.forEach(input => input.value = '');
    }

    render() {
        return (
            <div className="col-3 offset-1 form-wrap">
                <h2>Add Blog</h2>
                <form className="form" onSubmit={this.addBlogHandler}>
                    <div className="form-group">
                        <label>Title</label>
                        <input type="text" className="form-control" ref={title => this.titleInput = title }/>
                    </div>
                    <div className="form-group">
                        <label>Text</label>
                        <input type="text" className="form-control" ref={text => this.textInput = text} />
                    </div>
                    <div className="form-group">
                        <label>Author</label>
                        <input type="text" className="form-control" ref={author => this.authorInput = author} />
                    </div>
                    <button type="submit" className="btn btn-primary">Add Post</button>
                </form>
            </div>
        )
    }
}

export default AddBlog