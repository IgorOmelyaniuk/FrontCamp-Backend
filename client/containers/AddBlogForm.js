import React from 'react';
import { Field, reduxForm } from 'redux-form';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { addBlog } from '../actions';

class AddBlogForm extends React.Component {
    
  renderField(field) {
    const { meta: {touched, error} } = field;
    const className = `form-group ${touched && error ? 'has-danger' : ''}`;

    return (
      <div className={className}>
        <label>{field.label}</label>
        <input
          className="form-control"
          type="text"
          {...field.input}
        />
        <div className="text-help">{touched ? error : ''}</div>
      </div>
    )
  }

  onSubmit(values) {
    this.props.addBlog(values, () => this.props.history.push('/'));
  }

  render() {
    const { handleSubmit } = this.props;

    return (
      <div className="col-6 offset-3 form-wrap">
        <h2 className="text-center">Add Blog</h2>
        <form onSubmit={handleSubmit(this.onSubmit.bind(this))}>
          <Field
            label="Title"
            name="title"
            component={this.renderField}
          />
          <Field
            label="Text"
            name="text"
            component={this.renderField}
          />
          <Field
            label="author"
            name="author"
            component={this.renderField}
          />
          <button style={{marginRight: '20px'}} type="submit" className="btn btn-primary">Add Post</button>
          <Link className="btn btn-danger" to="/">Cancel</Link>
        </form>
      </div>
    )
  }
}

const validate = values => {
  const errors = {};

  if (!values.title || values.title.length < 3) {
    errors.title = 'Enter a title that is at least 3 characters'
  }
  
  if (!values.text) {
    errors.text = 'Enter a text'
  }
  
  if (!values.author) {
    errors.author = 'Enter a author'
  }
    
  return errors;
}

export default reduxForm({
  validate,
  form: 'AddForm'
})(
  connect(null, { addBlog })(AddBlogForm)
)