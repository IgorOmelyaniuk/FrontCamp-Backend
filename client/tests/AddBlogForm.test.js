import React, { Component } from 'react';
import { shallow } from 'enzyme';
import AddBlogForm from '../components/AddBlogForm'; 
import toJson from 'enzyme-to-json'

describe('subscribe component', () => {
  it('should render AddBlogForm', () => {
    expect(toJson(shallow(<AddBlogForm/>))).toMatchSnapshot();
  });
});