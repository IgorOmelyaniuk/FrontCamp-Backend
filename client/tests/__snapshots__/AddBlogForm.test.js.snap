// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscribe component should render AddBlogForm 1`] = `
<Connect(Form(Connect(AddBlogForm)))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="AddForm"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
  validate={[Function]}
/>
`;
